### PR TITLE
Better integration of `ListItem` in tooltips and other popups

### PR DESCRIPTION
**Pull request by emilk**
### What
* Closes https://github.com/rerun-io/rerun/issues/6516

This PR lets us put any `ListItem` anywhere without trouble.

This is because of two things:
* `ListItem` understands the concept of an egui sizing pass, so won't expand to become huge when in a tooltip
* `LabelContent` and `PropertyContent` can tell when it is in a tooltip, and in these cases won't truncate the contents

image

image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6486?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6486?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6486)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/re_data_ui/src/instance_path.rs [hunk 1]
```diff
@@ -145,11 +145,6 @@ impl DataUi for InstancePath {
                                 }
                             });
 
-                    // avoid the list item to max the tooltip width
-                    if ui_layout == UiLayout::Tooltip {
-                        content = content.exact_width(true);
-                    }
```
Comment (emilk) [inline crates/re_data_ui/src/instance_path.rs:151]: This is now detected automagically by `PropertyContent`

File: crates/re_time_panel/src/lib.rs [hunk 2]
```diff
@@ -626,7 +626,7 @@ impl TimePanel {
                         ctx,
                         &InstancePath::from(tree.path.clone()),
                     ))
-                    .exact_width(true),
+                    .truncate(false),
```
Comment (emilk) [inline crates/re_time_panel/src/lib.rs:629]: We don't yet magically detect that we are in the time panel, so we still need to tell the contents not to truncate here

File: crates/re_data_ui/src/item_ui.rs [hunk 3]
```diff
@@ -417,8 +417,7 @@ pub fn component_path_button_to(
                 } else {
                     "Temporal component"
                 })
-                .with_icon(icon)
-                .exact_width(true),
```
Comment (emilk) [inline crates/re_data_ui/src/item_ui.rs:421]: again, the contents magically knows not to truncate when in a tooltip

File: crates/re_ui/src/lib.rs [hunk 4]
```diff
@@ -88,3 +88,29 @@ pub fn apply_style_and_install_loaders(egui_ctx: &egui::Context) {
 
     design_tokens().apply(egui_ctx);
 }
+
+/// Used as a heuristic to figure out if it is safe to truncate text.
+///
+/// If this returns false, we should never truncate.
```
Comment (jleibs) [inline crates/re_ui/src/lib.rs:94]: > If this returns false, we should never truncate.

Double-negative. Would be easier to follow as "If this returns true, we should truncate text when necessary."

The rationale coupling this description to the name of the function still isn't totally obvious. Even if a panel is not resizable, if the text is much longer than the available (non-resizable) panel size, it seems like truncation could be necessary.

**Review by jleibs — APPROVED** [review]

File: crates/re_ui/src/lib.rs (hunk 4 above)
Comment (emilk) [reply crates/re_ui/src/lib.rs]: I expanded the docstring to clarify

**Pull request merged**